_inc/client/components/jetpack-termination-dialog/dialog.jsx

```diff
@@ -0,0 +1,173 @@
+	switch ( benefitName ) {
+		case 'contact-form':
+			return 'align-image-center';
+		case 'contact-form-feedback':
+			return 'mail';
+		case 'image-hosting':
+			return 'image';
+		case 'jetpack-backup':
+			return 'cloud-download';
+		case 'jetpack-stats':
+			return 'stats-alt';
+		case 'protect':
+			return 'lock';
+		case 'publicize':
+			return 'share';
+		case 'sharing':
+			return 'share';
+		case 'subscribers':
+			return 'user';
+		case 'video-hosting':
+			return 'video-camera';
+		default:
+			return 'checkmark';
+	}
+}
+
+function mapBenefitDataToViewData( benefit ) {
+	return {
+		title: benefit.title,
+		description: benefit.description,
+		amount: benefit.value,
+		gridIcon: mapBenefitNameToGridicon( benefit.name ),
+	};
+}
+
+/*
+ * On Jetpack Termination
+ *
+ * This Dialog is designed to be used from multiple locations with different intents, either
+ * disconnecting the Jetpack plugin or uninstalling it. To abstract this we use the word
+ * "Termination" to represent both
+ */
+
+class JetpackTerminationDialog extends Component {
+	static propTypes = {
+		closeDialog: PropTypes.func.isRequired,
+		disconnectJetpack: PropTypes.func.isRequired,
+		location: PropTypes.oneOf( [ 'plugins', 'dashboard' ] ).isRequired,
+		purpose: PropTypes.oneOf( [ 'disconnect', 'disable' ] ).isRequired,
+		siteBenefits: PropTypes.array,
+		siteName: PropTypes.string,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.handleTerminationClick = this.handleTerminationClick.bind( this );
+		this.handleDialogCloseClick = this.handleDialogCloseClick.bind( this );
+	}
+
+	componentDidMount() {
+		const { location, purpose, siteName } = this.props;
+		analytics.tracks.recordEvent( 'jetpack_termination_dialog_termination_view', {
+			location,
+			purpose,
+			siteName,
+		} );
+	}
+
+	handleTerminationClick() {
+		const { location, purpose, siteName } = this.props;
+		analytics.tracks.recordEvent( 'jetpack_termination_dialog_termination_click', {
+			location,
+			purpose,
+			siteName,
+		} );
+		// TODO: re-enable before shipping
+		// this.props.terminateJetpack();
+	}
+
+	handleDialogCloseClick() {
+		const { location, purpose, siteName } = this.props;
+		analytics.tracks.recordEvent( 'jetpack_termination_dialog_close_click', {
+			location,
+			purpose,
+			siteName,
+		} );
+		this.props.closeDialog();
+	}
+
+	renderFeatures() {
+		const { siteBenefits, siteName } = this.props;
+
+		return siteBenefits ? (
+			<JetpackTerminationDialogFeatures
+				siteBenefits={ siteBenefits.map( mapBenefitDataToViewData ) }
+				siteName={ siteName }
+			/>
+		) : (
+			<Card className="jetpack-termination-dialog__spinner">
+				<Spinner size={ 50 } />
+			</Card>
+		);
+	}
+
+	render() {
+		const { purpose, location } = this.props;
+
+		return (
+			<div className="jetpack-termination-dialog">
+				<QuerySite />
+				<QuerySiteBenefits />
+				<Card>
+					<div className="jetpack-termination-dialog__header">
+						<h1>
```

Comment:
We use `h2` headings in other areas (like the Mobile login modal next to this one). Maybe we should use `h2` here as well.